player.cpp

```diff
@@ -755,11 +782,61 @@ void test_player() //!OCLINT tests may be long
     assert(p.get_position() == c);
   }
 #endif
-  #ifdef FIX_ISSUE_351
+#ifdef FIX_ISSUE_351
   {
     assert(to_str(player_state::active) == "active");
   }
-  #endif
+#endif
+//#define FIX_ISSUE_367
+#ifdef FIX_ISSUE_367
+  {
+    {
+      player p;
+      const coordinate starting_position = p.get_position();
+      assert(p.get_speed() < 0.0000000001 && p.get_speed() > -0.00000000001);
+      p.move();
+      assert(starting_position == p.get_position());
+      p.turn_left();
+      assert(starting_position == p.get_position());
+    }
+    {
+      player p;
+      const coordinate starting_position = p.get_position();
+      assert(p.get_speed() < 0.0000000001 && p.get_speed() > -0.00000000001);
+
+      // a player with direction 0 and speed 1 moves one unit of space along the x-axis
+      while(p.get_speed() < 1) {
```

Comment:
> make sure that acceleration is always a multiple of 1

Isn't that true for any number 😜
Did you mean `assert(1 % p.get_acceleration() == 0 && p.get_acceleration()  <= 1)`

Reply:
> Did you mean assert(1 % p.get_acceleration() == 0 && p.get_acceleration() <= 1)
This is equvialent to what I wrote I suspect.

> What did you mean by the <= 1 condition defeat the purpose of != 1 ?
The purpose of `!=1` is to make sure that you end the loop being _exactly_ on 1 and not with just a value higher than 1. This is because for the test to work you need to make sure your speed is 1. Since we do not have and do not want to have a `set_speed()`, function I figured this was the best way. An alternative I might suggest is to write a long_named_function such as `player::set_speed_to_a_given_value_for_test(double speed)` so that we use it for tests only, to set a precise speed, but nobody is tempted to use for the actual code.

Reply:
so, the annoying thing about modulo is that it olny works with integers.

Reply:
how about this:
```      
while(p.get_speed() <= 1) {
          p.accelerate();
}
assert(p.get_speed() < 1.00000000000001 && p.get_speed() > 0.999999999999999);
```